js/ui/components/networkAgent.js

```diff
@@ -162,11 +267,52 @@ var NetworkSecretDialog = new Lang.Class({
 
         if (valid) {
             this._agent.respond(this._requestId, Shell.NetworkAgentResponse.CONFIRMED);
+
+            // Update the Automatic Updates values
+            if (this._settingName != 'vpn')
+                this._updateMeteredSetting(this._updatesToggle.checked);
+
             this.close(global.get_current_time());
         }
         // do nothing if not valid
     },
 
+    _updateMeteredSetting: function(isMeteredConnection) {
+        // We cannot change the connection here (it's just a NMSimpleConnection
+        // and it'll discard all our settings after destroyed) so we're forced to
+        // use the NMRemoteConnection that the current connection represents.
+        let remoteConnection = this._client.get_connection_by_path(this._connection.get_path());
+
+        if (!remoteConnection)
+            return;
+
+        // Store the Automatic Updates first
+        let userSetting = remoteConnection.get_setting(NM.SettingUser.$gtype);
+
+        if (!userSetting) {
+            userSetting = new NM.SettingUser();
+            remoteConnection.add_setting(userSetting);
+        }
+
+        let value = isMeteredConnection ? '0' : '1';
```

Comment:
You don't need to define a var for something that'll just be used inside the next line, right?

Reply:
Well, I just thought it'd make the following line shorter. No strong opinion here.